control-plane/go.mod

```diff
@@ -49,6 +47,11 @@ require (
 	sigs.k8s.io/gateway-api v0.7.1
 )
 
+require (
+	github.com/hashicorp/consul-k8s/control-plane/cni v0.0.0-20230905204616-c6b703d1deff
+	github.com/hashicorp/consul-server-connection-manager v0.1.4
+)
```

Comment:
~ Maybe just me not understanding `go mod tidy` magic: why the new `require` block here? I'm used to only seeing direct and indirect deps split out.

Reply:
Yeah, not sure. I moved to 1.21 locally, and I've seen a bit of this moving around. I didn't see anything in the release notes about it. I'll probably leave it for now and assume `go mod tidy` knows what it likes.

Reply:
Sounds good; CI should complain if `1.20` `go mod tidy` doesn't like the format.